autoscheduler/frontend/src/redux/actions/courseCards.ts

```diff
@@ -36,57 +36,98 @@ function updateCourseCardSync(index: number, courseCard: CourseCardOptions): Upd
+ * @param section The parsed Section type from this sectionData
+ */
+function parseMeetings(sectionData: any, section: Section): Meeting[] {
+  const meetings: Meeting[] = [];
+
+  sectionData.meetings.forEach((meetingData: any) => {
+    let start: string[] = ['0', '0'];
+    let end: string[] = ['0', '0'];
+
+    if (meetingData.start_time != null && meetingData.start_time.length > 0) {
+      start = meetingData.start_time.split(':');
+      end = meetingData.end_time.split(':');
+    }
+
+    const meeting = new Meeting({
+      id: Number(meetingData.id),
+      building: '',
+      meetingDays: meetingData.days,
+      startTimeHours: Number(start[0]),
+      startTimeMinutes: Number(start[1]),
+      endTimeHours: Number(end[0]),
+      endTimeMinutes: Number(end[1]),
+      // Converts string to MeetingType enum
+      meetingType: MeetingType[meetingData.type as keyof typeof MeetingType],
+      section,
+    });
+
+    meetings.push(meeting);
+  });
+
+  return meetings;
+}
+
+/**
+ * Helper function that parses all of the meetings in the section data and returns them.
+ * Used in ConfigureCard for parsing the return from /scheduler/generate
+ * @param arr The array of sections returne from the backend
+ */
+export function parseAllMeetings(arr: any[]): Meeting[] {
+  const ret: Meeting[] = [];
+
+  arr.forEach((sectionData) => {
+    const section = parseSection(sectionData);
+
+    const meetings = parseMeetings(sectionData, section);
+
+    ret.push(...meetings);
+  });
+
+  return ret;
+}
+
 /**
  *  Helper function to parse the serialized Section model from the backend convert it into an
  *  array of Sections
  *  @param arr The array of sections returned from the backend, such as from api/sections
  */
-export function parseSections(arr: any[]): SectionSelected[] {
+export function parseSections(arr: any[]): SectionSelected[] { // Rename to parseSectionSelected?
   const sections: SectionSelected[] = [];
 
   arr.forEach((sectionData) => {
-    const meetings: Meeting[] = [];
-
-    const section = new Section({
-      id: Number(sectionData.id),
-      crn: Number(sectionData.crn),
-      subject: sectionData.subject,
-      courseNum: sectionData.course_num,
-      sectionNum: sectionData.section_num,
-      minCredits: Number(sectionData.min_credits),
-      maxCredits: Number(sectionData.max_credits) || null,
-      currentEnrollment: Number(sectionData.current_enrollment),
-      maxEnrollment: Number(sectionData.max_enrollment),
-      honors: sectionData.honors,
-      web: sectionData.web,
-      instructor: new Instructor({ name: sectionData.instructor_name }),
-      grades: sectionData.grades == null ? null : new Grades(sectionData.grades),
-    });
-
-    sectionData.meetings.forEach((meetingData: any) => {
-      let start: string[] = ['0', '0'];
-      let end: string[] = ['0', '0'];
+    const section = parseSection(sectionData);
```

Comment:
So basically, these two functions have been separated out into helper functions, and those helper functions are re-used between `parseSections` and `parseAllMeetings`, right?

Reply:
Yeah exactly. They're only very slightly different and since they're both needed for different situations, I thought extracting out the common functionality was the best move